src/lazy/expanded/mod.rs

```diff
@@ -178,7 +178,7 @@ impl EncodingContext {
     }
 
     pub fn get_ref(&self) -> EncodingContextRef<'_> {
-        EncodingContextRef { context: self }
+        EncodingContextRef { context: self, value_start: 0, value_end: 0 }
```

Comment:
This looks like `value_start` and `value_end` do not belong in `EncodingContextRef`.

If `value_start` and `value_end` are part of an `EncodingContextRef`, then you should be able to set them based on some value in `EncodingContext` instead of defaulting to `0`, but if they are part of `EncodingContext`, then these fields are unnecessary because they are already included via the `context` field.

Reply:
Thats a good point. I will change that to just make `location()` take in `value_start` and `value_end` instead of setting it up this way.